Jenkinsfile

```diff
@@ -24,68 +24,120 @@ pipeline {
 
     // Right now, we're *only* building and deploying on the `master` branch;
     // longer-term, we'll want to deploy feature branches as well.
-    stage('`master` branch') {
+
+    stage('Build base Docker image') {
       when { branch 'master' }
-      stages {
+      agent any
+      steps {
+        script {
+          def dockerRepoName = 'zooniverse/front-end-monorepo'
+          def dockerImageName = "${dockerRepoName}:${BRANCH_NAME}"
+          def newImage = docker.build(dockerImageName, "--target bootstrap ./")
+          newImage.push()
+          newImage.push('latest')
+        }
+      }
+    }
+
+    stage('Build production app images') {
+      when { tag 'production-release' }
```

Comment:
It might be simpler to collapse prod and master to the same stage here using when `anyOf` https://jenkins.io/doc/book/pipeline/syntax/#when-example

then detect the presence of the TAG_NAME to determine if it's prod build vs staging?

Reply:
That's an excellent idea. Have you memorised all the Jenkins documentation? 😉

Reply:
So I could go back to using a ternary based on `TAG_NAME`?

Reply:
i think so - best to test. 

I tend to use the replay jobs function of jenkins to test jenkinsfiles, https://pushbuildtestdeploy.com/using-jenkinss-replay-feature-to-work-on-without-repeatedly-committing-the-jenkinsfile/

Reply:
If it doesn't work, I think the worst that happens is `APP_ENV` is staging for all builds. As far as I can tell, it's only used to set the Sentry event environment, and we aren't using those in Sentry anyway. So it wouldn't improve the error reports, but it wouldn't break anything either.

Reply:
I replayed a build for this branch, changing `branch 'master'` to `branch 'sentry-environments'` throughout the Jenkinsfile. It still skips those `when` conditions and I can't see why.
```
Stage "Build app images" skipped due to when conditional
```
https://jenkins.zooniverse.org/job/Zooniverse%20GitHub/job/front-end-monorepo/job/PR-1589/7/console

Maybe I need to check a different value of the branch name for a PR merge?

Reply:
With the correct branch name, Jenkins builds and deploys to staging, and I'm seeing the `staging` environment tagged on events from that build.